app/exerlog/lib/UI/calendar/widgets/logout_button.dart

```diff
@@ -24,7 +25,46 @@ class _LogoutButtonState extends State<LogoutButton> {
             setState(() {
               _isLoading = true;
             });
-            await FirebaseAuth.instance.signOut();
+            bool shouldLogout = false;
```

Comment:
Try to separate the logic from the ui. Ideally we would do this in a viewModel, but for now you can just move this function from inside of the widget to the bottom outside the build function, and then just call it here when pressing the inkwell. 

```
Inkwell(
onTap: _onLogoutTapped,
child: // the rests of the widget
)

void _onLogoutTapped(){
  //implementation of logout function
 }
```